### PR TITLE
Add dive rating and visibility to statistics variables

### PR DESCRIPTION
**Pull request by atdotde**
Signed-off-by: Robert C. Helling <helling@atdotde.de>

### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

File: stats/statsvariables.cpp [hunk 1]
```diff
@@ -1681,6 +1681,58 @@ struct DayOfWeekVariable : public StatsVariableTemplate<StatsVariable::Type::Dis
...
+	QString diveCategories(const dive *d) const override {
+		int rating = (int)d->rating;
+		return QString("🌟").repeated(rating);
+	}
+	std::vector<const StatsBinner *> binners() const override {
+		return { &rating_binner };
+	}
+};
+
+// ============ Rating ============
```
Comment (bstoeger) [inline stats/statsvariables.cpp]: Visibility

File: stats/statsvariables.cpp [hunk 2]
```diff
@@ -1681,6 +1681,58 @@ struct DayOfWeekVariable : public StatsVariableTemplate<StatsVariable::Type::Dis
...
+struct VisibilityBinner : public SimpleBinner<RatingBinner, IntBin> {
```
Comment (bstoeger) [inline stats/statsvariables.cpp]: I'm surprised that this compiles. This should read VisibilityBinner. I will have to add some TMP (template meta programming) to catch this kind of error.

**Review by bstoeger — APPROVED** [review]
Two small copy & paste errors. If this compiles and works, it is fine with me.

One could think about a common base class or template for all the "star" variables, but it's probably not worth it. On a rainy afternoon maybe.

I really should remember to catch that wrong template argument, though.

**Comment by dirkhh** [comment]
> Two small copy & paste errors. If this compiles and works, it is fine with me.
> 
> One could think about a common base class or template for all the "star" variables, but it's probably not worth it. On a rainy afternoon maybe.

BTW, not suggesting that this should be added, but we do have more of those star values. Waves, surge, and chill IIRC

> I really should remember to catch that wrong template argument, though.

Oh that will be fun. I can't wait to read THAT code 😉

**Comment by atdotde** [comment]
Fixed the c&p errors. Decided against surf until somebody complains ;-)

**Pull request merged**